public/app/features/serviceaccounts/ServiceAccountsTable.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { FC, useEffect, useState } from 'react';
+import { AccessControlAction, Role, OrgServiceaccount } from 'app/types';
+import { OrgRolePicker } from '../admin/OrgRolePicker';
+import { Button, ConfirmModal } from '@grafana/ui';
+import { OrgRole } from '@grafana/data';
+import { contextSrv } from 'app/core/core';
+import { fetchBuiltinRoles, fetchRoleOptions, UserRolePicker } from 'app/core/components/RolePicker/UserRolePicker';
+
+export interface Props {
+  serviceaccounts: OrgServiceaccount[];
+  orgId?: number;
+  onRoleChange: (role: OrgRole, serviceaccount: OrgServiceaccount) => void;
+  onRemoveServiceaccount: (serviceaccount: OrgServiceaccount) => void;
+}
+
+const ServiceaccountsTable: FC<Props> = (props) => {
+  const {
+    serviceaccounts: serviceaccounts,
+    orgId,
+    onRoleChange,
+    onRemoveServiceaccount: onRemoveserviceaccount,
+  } = props;
+  const canUpdateRole = contextSrv.hasPermission(AccessControlAction.OrgUsersRoleUpdate);
+  const canRemoveFromOrg = contextSrv.hasPermission(AccessControlAction.OrgUsersRemove);
+  const rolePickerDisabled = !canUpdateRole;
+
+  const [showRemoveModal, setShowRemoveModal] = useState<string | boolean>(false);
```

Comment:
The usage of `string | boolean` for this state looks a bit confusing imho. I see you use the `serviceaccount.login` field (which is a string) to determine if that modal should be displayed or not. So, I'm wondering if we could simplify that logic by using just a boolean to determine it and set the value of that boolean based on `serviceaccount.login` value (and update it every time that value is updated as well) instead of using both within the state meant to determine if the modal is displayed or not.

However, I'm not an experienced TS developer, so there could be a fair reason to mix them up 🤷🏻‍♀️

Reply:
good catch @joanlopez, I've missed this one. Indeed, we should not mix the types here, and just use bool value.

Reply:
Fixed that in 6cdcec275701788cc0b95e4be9359f2eb4e5d408